source/persistency/PetaloPersistencyManager.cc

```diff
@@ -285,114 +284,49 @@ void PetaloPersistencyManager::StoreIonizationHits(G4VHitsCollection* hc)
-    std::map<G4double, G4int>::const_iterator it;
-
-    if (save_tot_charge_ == true) {
-      G4double charge = 0.;
-      for (it = wvfm.begin(); it != wvfm.end(); ++it) {
-        charge = charge + (*it).second;
+    if (charge > thr_charge_){
+      std::string sdname = hits->GetSDname();
+      G4ThreeVector xyz = hit->GetPosition();
+      if (save_tot_charge_ == true) {
+        h5writer_->WriteSensorDataInfo(nevt_, (unsigned int)s_id, (unsigned int)charge);
       }
-        h5writer_->WriteSensorDataInfo(nevt_, (unsigned int)hit->GetPmtID(), charge);
-    }
-
-    if (hit->GetPmtID() >= 0) {
       std::vector<G4int>::iterator pos_it =
-	std::find(sns_posvec_.begin(), sns_posvec_.end(), hit->GetPmtID());
+        std::find(sns_posvec_.begin(), sns_posvec_.end(), s_id);
       if (pos_it == sns_posvec_.end()) {
-	h5writer_->WriteSensorPosInfo((unsigned int)hit->GetPmtID(), sdname.c_str(),
+        h5writer_->WriteSensorPosInfo((unsigned int)s_id, sdname.c_str(),
                                       (float)xyz.x(), (float)xyz.y(), (float)xyz.z());
-	sns_posvec_.push_back(hit->GetPmtID());
-      }
-    }
-
-
-    // TOF
-    SensorHit* hitTof = mapOfHits[-s_id];
-    const std::map<G4double, G4int>& wvfmTof = hitTof->GetHistogram();
-
-    double binsize_tof = hitTof->GetBinSize();
-
-    for (it = wvfmTof.begin(); it != wvfmTof.end(); ++it) {
-
-      if (((*it).first) <= tof_time_){
-        unsigned int time_bin_tof = (unsigned int)((*it).first/binsize_tof+0.5);
-        unsigned int charge_tof = (unsigned int)((*it).second+0.5);
-        h5writer_->WriteSensorTofInfo(nevt_, hitTof->GetPmtID(), time_bin_tof, charge_tof);
+        sns_posvec_.push_back(s_id);
       }
-      else {
-        break;
-      }
-    }
-
-    /*
-    const std::map<G4double, G4double>&  wvls= hit->GetWavelengths();
-    std::map<G4double, G4double>::const_iterator w;
-
-    std::vector<double > value;
-    int count =0;
-    std::ostringstream strs;
-    strs << hit->GetPmtID();
-    std::string sens_id = strs.str();
-    //   G4cout << "Longitud = " << wvls.size() << G4endl;
-    for (w = wvls.begin(); w != wvls.end(); ++w) {
-      if (count < 100) {
-	value.clear();
-	std::ostringstream strs2;
-	strs2 << count;
-	std::string order = strs2.str();
-	std::string key = sens_id + '_' + order;
-	value.push_back(w->first/CLHEP::picosecond);
-	value.push_back(w->second/CLHEP::nanometer);
-	//	G4cout << key << ", " << value[0] << ", " << value[1] << G4endl;
-	ievt->fstore(key, value);
-	count++;
+      // Save also individual photons
+      const std::map<G4double, G4int>& phot = hit->GetPhotonMap();
+      std::map<G4double, G4int>::const_iterator it;
+      for (it = phot.begin(); it != phot.end(); ++it) {
+        if (it->first <= tof_time_){
+          h5writer_->WriteSensorTofInfo(nevt_, (unsigned int)s_id, (float)it->first,
+                                        (unsigned int)it->second);
+        }
```

Comment:
I see that in table `tof_response`, the time for each photon is saved in ns. I don't think it matters but did you choose it on purpose?

Reply:
In the rest of tables we save the quantities with the internal units of Geant4, therefore I would do the same for `tof_response` (which, for time, is nanoseconds). I think that it is less prone to confusion if we keep the same convention everywhere.